src/AnalyticsEventReporter.ts

```diff
@@ -60,22 +60,19 @@ export class AnalyticsEventReporter implements AnalyticsEventService {
       const shouldUseBeacon = useBeacon(finalPayload, this.config.forceFetch);
       const requestUrl = setupRequestUrl(this.config.env, this.config.region);
 
-      // If useBeacon returns true, return boolean response of postWithBeacon
+      // If useBeacon returns true, return boolean response of postWithBeacon as string.
       if (shouldUseBeacon) {
-        return await postWithBeacon(requestUrl, finalPayload);
+        return String(postWithBeacon(requestUrl, finalPayload));
       }
 
       /** If useBeacon returns false, use postWithFetch.
           If result is successful, return result json.
           If request fails, return errors. */
       const res = await postWithFetch(requestUrl, finalPayload);
```

Comment:
I realized that this function doesn't really send the data asynchronously as we are doing `await` on the post call.

Instead can we try to chain the promises so that we just return the chained promise. Something like 

```
const postPromise = postWithFetch(..)
postPromise.then((response) => {
  response.json()['id']
})
.catch(...)
```

That way the post report call is truly async